realm/realm-library/src/main/java/io/realm/internal/fields/FieldDescriptor.java

```diff
@@ -286,7 +287,9 @@ protected final void setCompilationResults(
         if (lastDotIndex > -1) {
             return Arrays.asList(FIELD_SEPARATOR.split(fieldDescription));
         } else {
-            return Collections.emptyList();
+            ArrayList<String> result = new ArrayList<>(1);
```

Comment:
Is it possible to use `return Collections.singletonList(fieldDescription)` here? Or does this have to be mutable?

Reply:
No, it doesn't. I didn't know that class existed, nice 👍